wandb/sdk/internal/job_builder.py

```diff
@@ -137,7 +135,7 @@ def _handle_server_artifact(
                 self._job_seq_id = res["artifactSequence"]["id"]
             except KeyError as e:
                 _logger.info(f"Malformed response from ArtifactSaver.save {e}")
-        if artifact.type == "code" and res is not None:
+        if artifact.type == "code" and "id" in res:
```

Comment:
Assuming this is a safe change

Reply:
Yep, the type changed from `Optional[Dict]` to just `Dict` which means the access in the line below is unsafe. This check should make it safer.